src/controllers/wallet/wallet.controller.js

```diff
@@ -763,6 +763,42 @@ async function stop(req, res) {
   res.send({ success: true });
 }
 
+/**
+ * Mark the inputs from the txHex as used.
+ *
+ * @param {import('express').Request} req
+ * @param {import('express').Response} res
+ */
+async function markUtxosSelectedAsInput(req, res) {
+  const validationResult = parametersValidation(req);
+  if (!validationResult.success) {
+    res.status(400).json(validationResult);
+    return;
+  }
+
+  /** @type {{ wallet: import('@hathor/wallet-lib').HathorWallet }} */
+  const { wallet } = req;
+  const { txHex, ttl } = req.body;
+  const markAsUsed = req.body.mark_as_used ?? true;
+
+  try {
+    const tx = helpersUtils.createTxFromHex(txHex, wallet.getNetworkObject());
+    await Promise.all(
+      tx.inputs.map(
+        input => (wallet.storage.utxoSelectAsInput(
```

Comment:
[note] I believe there is an issue in the `utxoSelectAsInput`:

```js
  /**
   * Mark an utxo as selected as input
   *
   * @param {IUtxoId} utxo The Data to identify the utxo
   * @param {boolean} markAs Mark the utxo as this value
   * @param {number|undefined} ttl Unmark the utxo after this amount os ms passed
   *
   * @async
   * @returns {Promise<void>}
   */
  async utxoSelectAsInput(utxo: IUtxoId, markAs: boolean, ttl?: number): Promise<void> {
    const tx = await this.getTx(utxo.txId);
    if (!tx || !tx.outputs[utxo.index]) {
      return;
    }

    if (markAs && tx.outputs[utxo.index].spent_by !== null) {
      // Already spent, no need to mark as selected_as_input
      return;
    }

    const utxoId = `${utxo.txId}:${utxo.index}`;
    if (markAs) {
      this.utxosSelectedAsInput.set(utxoId, markAs);
      // if a ttl is given, we should reverse
      if (ttl) {
        setTimeout(() => {
          if (!markAs) {
            if (this.utxosSelectedAsInput.has(utxoId)) {
              this.utxosSelectedAsInput.delete(utxoId);
            }
          }
        }, ttl);
      }
    } else {
      this.utxosSelectedAsInput.delete(utxoId);
    }
  }
```

If `markAs` is `false` or `undefined` (I know we are validating to never let it be `undefined` from headless), we will delete the utxo as selected input, but we will not restore its previous state if we use it with `ttl` because there is no `setTimeout` in the `else` branch. Is it an issue? Do we want this behavior of ignoring `ttl` in case `markAs` is `false`?

Reply:
This is not a bug, we intentionally do not revert an unmark as "selected_as_input".

But while checking the method I found a bug on the method, created another [PR](https://github.com/HathorNetwork/hathor-wallet-lib/pull/773) to fix this

Reply:
Oh, ok!